uxarray/grid/geometry.py

```diff
@@ -1561,3 +1568,234 @@ def inverse_stereographic_projection(x, y, central_lon, central_lat):
     )
 
     return lon, lat
+
+
+@njit(cache=True)
+def point_in_face(
+    edges_xyz,
+    point_xyz,
+    inclusive=True,
+):
+    """Determines if a point lies inside a face.
+
+    Parameters
+    ----------
+        edges_xyz : numpy.ndarray
+            Cartesian coordinates of each point in the face
+        point_xyz : numpy.ndarray
+            Cartesian coordinate of the point
+        inclusive : bool
+            Flag to determine whether to include points on the nodes and edges of the face
+
+    Returns
+    -------
+    bool
+        True if point is inside face, False otherwise
+    """
+
+    # Validate the inputs
+    if len(edges_xyz[0][0]) != 3:
+        raise ValueError("`edges_xyz` vertices must be in Cartesian coordinates.")
+
+    if len(point_xyz) != 3:
+        raise ValueError("`point_xyz` must be a single [3] Cartesian coordinate.")
+
+    # Initialize the intersection count
+    intersection_count = 0
+
+    # Set to hold unique intersections
+    unique_intersections = set()
+
+    location = _classify_polygon_location(edges_xyz)
+
+    if location == 1:
+        ref_point_xyz = np.array([0.01745241, 0.0, -0.9998477], dtype=np.float64)
+    elif location == -1:
+        ref_point_xyz = np.array([0.01745241, 0.0, 0.9998477], dtype=np.float64)
+    else:
+        ref_point_xyz = np.array([0.01745241, 0.0, -0.9998477], dtype=np.float64)
```

Comment:
Please define these as constants at the top of the file to avoid the use of magic numbers.